src/setup_payload/QRCodeSetupPayloadParser.cpp

```diff
@@ -376,13 +376,15 @@ CHIP_ERROR QRCodeSetupPayloadParser::populatePayload(SetupPayload & outPayload)
     static_assert(kProductIDFieldLengthInBits <= 16, "Won't fit in uint16_t");
     outPayload.productID = static_cast<uint16_t>(dest);
 
-    err = readBits(buf, indexToReadFrom, dest, kCustomFlowRequiredFieldLengthInBits);
+    err = readBits(buf, indexToReadFrom, dest, kCommissioningFlowFieldLengthInBits);
     SuccessOrExit(err);
-    static_assert(kCustomFlowRequiredFieldLengthInBits <= 8, "Won't fit in uint8_t");
-    outPayload.requiresCustomFlow = static_cast<uint8_t>(dest);
+    static_assert(kCommissioningFlowFieldLengthInBits <= 8 * sizeof(CommissioningFlow), "Won't fit in CommissioningFlow");
```

Comment:
Perhaps
```suggestion
    static_assert(kCommissioningFlowFieldLengthInBits <= std::numeric_limits<std::underlying_type_t<CommissioningFlow>>::digits, "Won't fit in CommissioningFlow");
```
to be very foolproof.